src/azure-cli/azure/cli/command_modules/backup/custom_wl.py

```diff
@@ -0,0 +1,710 @@
+
+    # Trigger register and wait for completion
+    result = sdk_no_wait(True, client.register,
+                         vault_name, resource_group_name, fabric_name, container_name, param)
+    return cust_help._track_register_operation(cmd.cli_ctx, result, vault_name, resource_group_name, container_name)
+
+
+def re_register_wl_container(cmd, client, vault_name, resource_group_name, workload_type, container_name, container_type):
+    workload_type = workload_type_map[workload_type]
+
+    if not cust_help._is_native_name(container_name):
+        raise CLIError(
+            """
+            Container name passed cannot be a friendly name.
+            Please pass a native container name.
+            """)
+
+    containers = list_wl_containers(backup_protection_containers_cf(cmd.cli_ctx), resource_group_name, vault_name, container_type)
+    source_resource_id = None
+
+    for container in containers:
+        if container.name == container_name:
+            source_resource_id = container.properties.source_resource_id
+            break
+
+    if not source_resource_id:
+        raise CLIError(
+            """
+            No such registered container exists.
+            """)
+
+    properties = AzureVMAppContainerProtectionContainer(backup_management_type=container_type, workload_type=workload_type,
+                                                        operation_type='Reregister', source_resource_id=source_resource_id)
+    param = ProtectionContainerResource(properties=properties)
+
+    # Trigger register and wait for completion
+    result = sdk_no_wait(True, client.register,
+                         vault_name, resource_group_name, fabric_name, container_name, param)
+    return cust_help._track_register_operation(cmd.cli_ctx, result, vault_name, resource_group_name, container_name)
+
+
+def unregister_wl_container(cmd, client, vault_name, resource_group_name, container_name):
+    if not cust_help._is_native_name(container_name):
+        raise CLIError(
+            """
+            Container name passed cannot be a friendly name.
+            Please pass a native container name.
+            """)
+
+    # Trigger unregister and wait for completion
+    result = sdk_no_wait(True, client.unregister,
+                         vault_name, resource_group_name, fabric_name, container_name)
+    return cust_help._track_register_operation(cmd.cli_ctx, result, vault_name, resource_group_name, container_name)
+
+
+def show_wl_container(cmd, name, resource_group_name, vault_name, container_type, status="Registered"):
+    return protection_containers_cf(cmd.cli_ctx).get(vault_name, resource_group_name, fabric_name, name)
+
+
+def list_wl_containers(client, resource_group_name, vault_name, container_type, status="Registered"):
+    return cust_help._get_containers(client, container_type, status, resource_group_name, vault_name)
+
+
+def show_wl_item(cmd, resource_group_name, vault_name, container_name, name):
```

Comment:
can we add a check for native name here?

Reply:
Done